Realm/Tests/ObjectTests.m

```diff
@@ -683,6 +683,19 @@ - (void)testStringSizeLimits {
     [realm commitWriteTransaction];
 }
 
+- (void)testAddingObjectNotInSchemaThrows {
+    RLMRealmConfiguration *configuration = [RLMRealmConfiguration defaultConfiguration];
+    configuration.objectClasses = @[StringObject.class];
+    RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:nil];
+
+    [realm beginWriteTransaction];
+    RLMAssertThrowsWithReasonMatching([realm addObject:[[IntObject alloc] initWithValue:@[@1]]], @"Object type 'IntObject' not persisted in Realm");
```

Comment:
But then it will get confusing when they just try and use an object property that isn't an `RLMObject` subclass (i.e. NSArray) -- the error then will have nothing to do with the `objectClasses` array.